libraries/app/database_api.cpp

```diff
@@ -1763,6 +1771,101 @@ uint64_t database_api_impl::get_son_count()const
    return _db.get_index_type<son_index>().indices().size();
 }
 
+//////////////////////////////////////////////////////////////////////
+//                                                                  //
+// Sidechain Accounts                                               //
+//                                                                  //
+//////////////////////////////////////////////////////////////////////
+
+vector<optional<sidechain_address_object>> database_api::get_all_sidechain_addresses()const
+{
+   return my->get_all_sidechain_addresses();
+}
+
+vector<optional<sidechain_address_object>> database_api_impl::get_all_sidechain_addresses()const
```

Comment:
Please note that a call like this in thousands of addresses in the object will just return all available. A call like this should have a limit. How to get results past the limit then ? With pagination, ideally this call(and probably others) should have at least a `start` and `limit` parameters. `limit` cant never be greater than a hardcoded or configuration value.

In Bitshares api calls haves a configurable limit, this is relatively recent, in the past there was a hardcoded limit.

This arguments can be optional, we can use default values if none of them are present but we will never be returning the whole table. As they can be optional i suppose it can be fine to add them in a later stage. 

In order to allow optional arguments to api calls there is a fix needed to fc: https://github.com/bitshares/bitshares-fc/pull/126

Reply:
I will remove it, since I used it only as a test command, and it is not actually required for SON work.